gui/main.js

```diff
@@ -422,11 +417,31 @@ const startSync = async () => {
   desktop.events.on('sync-state', state => {
     enqueueStateUpdate('sync-state', state)
   })
-  desktop.events.on('transfer-started', addFile)
-  desktop.events.on('transfer-copy', addFile)
-  desktop.events.on('transfer-move', async (info, old) => {
-    await addFile(info)
-    await removeFile(old)
+  desktop.events.on('transfer-started', doc => {
+    const info = fileInfo(doc, { transferred: 0 })
+    enqueueStateUpdate('syncing', info)
+  })
+  desktop.events.on('transfer-progress', (doc, { transferred }) => {
+    const info = fileInfo(doc, { transferred })
+    enqueueStateUpdate('syncing', info)
+  })
+  desktop.events.on('transfer-done', doc => {
+    const info = fileInfo(doc)
+    enqueueStateUpdate('syncing', info)
+    addFile(doc)
+  })
+  desktop.events.on('transfer-failed', doc => {
+    const info = fileInfo(doc)
+    // XXX: No state update as it will come from a `sync-state` event
+    // TODO: find a way to have the old file info take its old place in the list
```

Comment:
TODO spotted

Reply:
I will leave this one for now because it's a very small issue that could take a while to get right.